Milestone 3: Game logic pull request

Albertino2020 opened the pull request:
In this milestone, the following features were completed:

1. - additional 2 files were added containing the Board Class and the Player's class, respectively
2. - classes were imported to the main object using require_relative
3. - game logic methods and functionalities were added
4. - the game asks players for their names
5. - the game tells which player turn it is
6. - the game asks the player to select from available moves
7. - the game informs player if the selected move is invalid
8. - the game displays board after the player move
9. - the game informs the player if the selected move is a winning move
10. - the game informs the player if the selected move is a draw move
11. - the game repeats all actions for the next player's move
12. - the game congratulates the winner
13. - the game displays final results.
14. - the game asks players if they want to continue playing and reads input options from the players
15. - the game restarts if players want to continue
16. - the game stops and players don't want to continue

lekegitrepo commented:
#### Hi, Good job :+1:  on the project, you did excellently.
#### However, you need to implement the following :point_down:  requirements before the project can be approved.

**_1. No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))_**

**_2. Use attr_readers/attr_writers instead of attr_accessors_**

**_Fix these and submit for review._**

#### _Reviewed by Temitayo Adeleke._

Albertino2020 commented:
> #### Hi, Good job  on the project, you did excellently.
> #### However, you need to implement the following  requirements before the project can be approved.
> **_1. No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))_**
> 
> **_2. Use attr_readers/attr_writers instead of attr_accessors_**
> 
> **_Fix these and submit for review._**
> 
> #### _Reviewed by Temitayo Adeleke._

Thank you, Temitayo,
As required, the following changes were done:
1- I removed puts/print and get.chomp from the Board and Player class and move them to the main.rb file

2. I replaced attr_accessors by attr_readers/attr_writers in the Board and Player classes.

Kind regards,
Albertino

Dandush03 commented on lib/player.rb:
```diff
@@ -0,0 +1,31 @@
+# rubocop: disable Style/GlobalVars
+class Player
+  attr_reader :name, :symbol, :turn, :turn_one, :turn_two, :choice, :winner
+  attr_writer :name, :symbol, :turn, :turn_one, :turn_two, :choice, :winner
+
```
#### I think this will be a good example

Notice how you used attr_reader/writer for the name variable, but after reviewing the code, I find out that you only set it up on your class constructor (initialize) and then only read from in other class,

The same example could be applied for more than just the name variable but I would let you dive in a do it by yourself to make sure you guys got the point

#### I hope it's all clear now 😉

Albertino2020 replied:
Hello Daniel,
I am really confused now.
I have made changes according to the previous TSE....and have to roll everything back??

When a player makes a winning move, the game stops. Didn't you see it?

Albertino2020 replied:
The game informs the player and stops. Then, it asks the players if they want to continue playing
Isn't it enough?
